airflow/providers/openlineage/plugins/listener.py

```diff
@@ -319,10 +322,46 @@ def on_failure():
                 len(Serde.to_json(redacted_event).encode("utf-8")),
             )
 
-        on_failure()
+        self._execute(on_failure, "on_failure", use_fork=True)
+
+    def _execute(self, callable, callable_name: str, use_fork: bool = False):
+        if use_fork:
+            self._fork_execute(callable, callable_name)
+        else:
+            callable()
+
+    def _fork_execute(self, callable, callable_name: str):
+        self.log.debug("Will fork to execute OpenLineage process.")
+        pid = os.fork()
+        if pid:
+            process = psutil.Process(pid)
+            try:
+                self.log.debug("Waiting for process %s", pid)
+                process.wait(conf.execution_timeout())
+            except psutil.TimeoutExpired:
+                self.log.warning(
+                    "OpenLineage process %s expired. This should not affect process execution.", pid
+                )
+                process.kill()
```

Comment:
I think we should add process.terminate() call as well and get another (possibly hard-coded) timeout before we send kill. Sending SIGKILL should generally be last-resort kinda thing - because it might leave some resources opened, locks locked etc. so we should really only do it if earlier SIGTERM did not terminate the process in reasonable time.

Reply:
Done - now we `process.terminate()` and give it 3 seconds to clean up. I think that's the reasonable timeout, and it should not really be configurable. Only then we try to `process.kill()`.